src/gt4py/stencil_object.py

```diff
@@ -167,118 +257,43 @@ def _call_run(self, field_args, parameter_args, domain, origin, exec_info=None):
-        for name, parameter in used_arg_params.items():
-            if not type(parameter) == self.parameter_info[name].dtype:
-                raise TypeError(
-                    f"The type of parameter '{name}' is '{type(parameter)}' instead of '{self.parameter_info[name].dtype}'"
-                )
 
-        assert isinstance(field_args, dict) and isinstance(parameter_args, dict)
-
-        # Shapes
-        shapes = {}
-
-        for name, field in used_arg_fields.items():
-            shapes[name] = Shape(field.shape)
         # Origins
         if origin is None:
             origin = {}
         else:
             origin = normalize_origin_mapping(origin)
-        for name, field in used_arg_fields.items():
+
+        for name, field in used_field_args.items():
             origin.setdefault(name, origin["_all_"] if "_all_" in origin else field.default_origin)
 
         # Domain
-        max_domain = Shape([sys.maxsize] * self.domain_info.ndims)
-        for name, shape in shapes.items():
-            upper_boundary = Index(self.field_info[name].boundary.upper_indices)
-            max_domain &= shape - (Index(origin[name]) + upper_boundary)
-
         if domain is None:
-            domain = max_domain
+            domain = Shape([sys.maxsize] * self.domain_info.ndims)
+            for name, field in used_field_args.items():
+                upper_boundary = Index(self.field_info[name].boundary.upper_indices)
+                domain &= Shape(field.shape) - (Index(origin[name]) + upper_boundary)
         else:
             domain = normalize_domain(domain)
-            if len(domain) != self.domain_info.ndims:
-                raise ValueError(f"Invalid 'domain' value '{domain}'")
 
-        # check domain+halo vs field size
-        if not domain > Shape.zeros(self.domain_info.ndims):
-            raise ValueError(f"Compute domain contains zero sizes '{domain}')")
-
-        if not domain <= max_domain:
-            raise ValueError(
-                f"Compute domain too large (provided: {domain}, maximum: {max_domain})"
-            )
-        for name, field in used_arg_fields.items():
-            min_origin = self.field_info[name].boundary.lower_indices
-            if origin[name] < min_origin:
-                raise ValueError(
-                    f"Origin for field {name} too small. Must be at least {min_origin}, is {origin[name]}"
-                )
-            min_shape = tuple(
-                o + d + h
-                for o, d, h in zip(
-                    origin[name], domain, self.field_info[name].boundary.upper_indices
-                )
-            )
-            if min_shape > field.shape:
-                raise ValueError(
-                    f"Shape of field {name} is {field.shape} but must be at least {min_shape} for given domain and origin."
-                )
+        if gt_config.debug:
+            self._validate_args(used_field_args, used_param_args, domain, origin, exec_info)
```

Comment:
The `max_domain` argument is not passed in, but is redefined on line 182 in `_validate_args`.

Reply:
Good catch! I removed the extra argument.